ts_src/payments/p2air.ts

```diff
@@ -0,0 +1,178 @@
+import { bitcoin as BITCOIN_NETWORK } from '../networks';
+import * as bscript from '../script';
+import * as vaultair from '../templates/vaultair';
+import { Payment, PaymentOpts, Stack, VaultTxType } from './index';
+import * as lazy from './lazy';
+
+const OPS = bscript.OPS;
+const typef = require('typeforce');
+const ecc = require('tiny-secp256k1');
+
+const OP_INT_BASE = OPS.OP_RESERVED; // OP_1 - 1
+
+function stacksEqual(a: Buffer[], b: Buffer[]): boolean {
+  if (a.length !== b.length) return false;
+
+  return a.every((x, i) => {
+    return x.equals(b[i]);
+  });
+}
+
+// input: OP_0 signature_1 [signature2, signature3] [OP_0/0x01] OP_0/0x01
+// output: OP_IF 1 OP_ELSE OP_IF 2 OP_ELSE 3 OP_ENDIF OP_ENDIF pubKey1 pubKey2 pubKey3 3 OP_CHECKMULTISIG
+export function p2air(a: Payment, opts?: PaymentOpts): Payment {
+  if (!a.input && !a.output && !a.pubkeys && !a.signatures)
+    throw new TypeError('Not enough data');
+  if (a.vaultTxType === undefined) a.vaultTxType = VaultTxType.Alert;
+
+  opts = Object.assign({ validate: true }, opts || {});
+
+  function isAcceptableSignature(x: Buffer | number): boolean {
+    return (
+      bscript.isCanonicalScriptSignature(x as Buffer) ||
+      (opts!.allowIncomplete && (x as number) === OPS.OP_0) !== undefined
+    );
+  }
+
+  typef(
+    {
+      network: typef.maybe(typef.Object),
+      output: typef.maybe(typef.Buffer),
+      pubkeys: typef.maybe(typef.arrayOf(ecc.isPoint)),
+      signatures: typef.maybe(typef.arrayOf(isAcceptableSignature)),
+      input: typef.maybe(typef.Buffer),
+      vaultTxType: typef.maybe(typef.Number),
+    },
+    a,
+  );
+
+  const network = a.network || BITCOIN_NETWORK;
+  const o: Payment = { network };
+
+  let chunks: Stack = [];
+  let decoded = false;
+  function decode(output: Buffer | Stack): void {
```

Comment:
the same as above with "as" usage. Also, as I see this function could be pure. So it would take output argument and return pubkeys. Then below you could use it like this

lazy.prop(o, 'pubkeys', () => {
    if (!a.output) return;
    return decode(a.output);
  });

Reply:
I see that in ar you also use this function, so it is another argument to make it pure and pass only the arguments for chunks.slice